.github/workflows/android.yml

```diff
@@ -47,15 +50,16 @@ jobs:
         if: ${{ endsWith(github.ref, '/staging') }}
         run: |
           echo "ENV=staging" >> $GITHUB_ENV
-          echo "SECRET_NAME=STAGING_ENV" >> $GITHUB_ENV
+          cat .vercel/.env.preview.local >> $GITHUB_ENV
           echo "APPCENTER_NAME=GoodDollar/GoodDollar-Android-staging" >> $GITHUB_ENV
           echo "APPCENTER_TOKEN=${{ secrets.APPCENTER_ANDROID_STAGING }}" >> $GITHUB_ENV
           echo "APPCENTER_CODEPUSH_TOKEN=${{ secrets.APPCENTER_CODEPUSH_STAGING }}" >> $GITHUB_ENV
       - name: Pre-checks - Env is PROD
         if: ${{ endsWith(github.ref, '/next') }}
         run: |
           echo "ENV=prod" >> $GITHUB_ENV
-          echo "SECRET_NAME=PROD_ENV" >> $GITHUB_ENV
+          vercel pull --yes --environment=production --token=${{ secrets.VERCEL_TOKEN }}
+          cat .vercel/.env.production.local >> $GITHUB_ENV
```

Comment:
Mobile actually uses the different .env.* files we have, so we need to overwrite them and not output to GITHUB_ENV

Reply:
Ah ye I see

Moved the pull to the later job and echoing to .env files.